### PR TITLE
Bump k8s dependencies to 1.22 and go to 1.16

### PR DESCRIPTION
**Pull request by nicolehanjing**
**What this PR does / why we need it**:
Bump k8s dependencies to 1.22 and go to 1.16
The merge is blocked by a kubebuilder issue:
Fix kubebuilder installation in workflow.
Caused by kubernetes-sigs/kubebuilder#2311
Fix was found here: https://github.com/operator-framework/operator-lifecycle-manager/pull/2335/files

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
Bump k8s dependencies to 1.22 and go to 1.16
```

### CODE REVIEW

**Comment by nicolehanjing** [comment]
@andrewsykim @lubronzhan can you give it a review when you get a chance? Thanks!

**Review by lubronzhan — APPROVED** [review]
/lgtm

File: go.mod [hunk 1]
```diff
@@ -1,35 +1,34 @@
...
-	k8s.io/apimachinery v0.21.0
-	k8s.io/client-go v0.21.0
-	k8s.io/cloud-provider v0.21.0
-	k8s.io/code-generator v0.21.0
-	k8s.io/component-base v0.21.0
-	k8s.io/klog/v2 v2.8.0
+	k8s.io/api v0.22.0
+	k8s.io/apiextensions-apiserver v0.22.0 // indirect
+	k8s.io/apimachinery v0.22.0
+	k8s.io/client-go v0.22.0
```
Comment (andrewsykim) [inline go.mod:27]: I _think_ 1.22.1 just came out, wondering if we should use that.

**Review by andrewsykim — COMMENTED** [review]
@nicolehanjing this PR looks good overall, can we create a release-1.21 branch off of master first before merging this?

File: go.mod (hunk 1 above)
Comment (nicolehanjing) [reply go.mod:27]: ah yes, let me update that

File: Makefile [hunk 2]
```diff
@@ -209,8 +209,8 @@ endif
 TEST_FLAGS ?= -v
 .PHONY: unit build-unit-tests
 unit unit-test:
-	curl -L https://go.kubebuilder.io/dl/2.3.1/${GOOS}/${GOARCH} | tar -xz -C /tmp/
-	mv /tmp/kubebuilder_2.3.1_${GOOS}_${GOARCH} /usr/local/kubebuilder
+	curl -L https://go.kubebuilder.io/dl/3.1.0/${GOOS}/${GOARCH} | tar -xz -C /tmp/
+	mv /tmp/kubebuilder_3.1.0_${GOOS}_${GOARCH} /usr/local/kubebuilder
```
Comment (nicolehanjing) [inline Makefile:213]: to pass CI tests, we need to update the kubebuilder version

File: Makefile [hunk 3]
```diff
@@ -209,8 +209,8 @@ endif
 TEST_FLAGS ?= -v
 .PHONY: unit build-unit-tests
 unit unit-test:
-	curl -L https://go.kubebuilder.io/dl/2.3.1/${GOOS}/${GOARCH} | tar -xz -C /tmp/
-	mv /tmp/kubebuilder_2.3.1_${GOOS}_${GOARCH} /usr/local/kubebuilder
+	curl -L https://github.com/kubernetes-sigs/kubebuilder/releases/download/v2.3.1/kubebuilder_2.3.1_${os}_${arch}.tar.gz | tar -xz -C /tmp/
+	sudo mv /tmp/kubebuilder_master_${GOOS}_${GOARCH} /usr/local/kubebuilder
```
Comment (nicolehanjing) [inline Makefile:213]: Fix kubebuilder installation in workflow.

Caused by kubernetes-sigs/kubebuilder#2311
Fix was found here: https://github.com/operator-framework/operator-lifecycle-manager/pull/2335/files

**Comment by nicolehanjing** [comment]
> @nicolehanjing this PR looks good overall, can we create a release-1.21 branch off of master first before merging this?

I created a release-1.21 branch just now - I should create this branch when cutting the release, but probably my git remote setting was not right so the release branch didn't get pushed

**Review by lubronzhan — APPROVED** [review]

**Pull request merged**